cfgrammar/src/lib/yacc/ast.rs

```diff
@@ -44,6 +46,33 @@ impl ASTWithValidityInfo {
         }
     }
 
+    /// Parses the `%grmtools section` expecting it to contain a `yacckind` entry.
+    pub fn from_src(src: &str) -> Result<Self, Vec<HeaderError>> {
```

Comment:
I named this `from_src` rather than `from_str` like LrLex is because for some reason clippy complains about this one, but not about LrLex, I assume because that one is a trait method?

Reply:
What's the Clippy error? It might be that we want to turn that one off (or maybe not!).

Reply:
Another option of course is implementing the `FromStr` trait.
https://rust-lang.github.io/rust-clippy/master/index.html#should_implement_trait

Here is  playground link in case you want to see it
https://play.rust-lang.org/?version=stable&mode=debug&edition=2024&gist=e9f0dd766fa27e37b9249db7150fc29e

Reply:
Good point, `FromStr` makes sense.

Reply:
This is partially done in [34906f9](https://github.com/softdevteam/grmtools/pull/554/commits/34906f9956caa755a76c183e741af5138ed90b37)

There is another `from_src` function that needs to move into `YaccGrammar<_>`, but is blocked on
adding generics to `HeaderError`.